plugin.go

```diff
@@ -78,14 +78,21 @@ func (h *IstioServiceEntries) readServiceEntries() {
 			continue
 		}
 
-		if entry.Resolution == networking.ServiceEntry_NONE || len(entry.Addresses) == 0 {
+		if entry.Resolution == networking.ServiceEntry_NONE {
 			// NO DNS based service discovery for service entries
 			// that specify NONE as the resolution. NONE implies
 			// that Istio should use the IP provided by the caller
 			continue
 		}
 
-		vips := convertToVIPs(entry.Addresses)
+		addresses := entry.Addresses
+		if len(addresses) == 0 {
+			// If the ServiceEntry has no Addresses, just map to a
+			// fake value; effectively minting a VIP
+			addresses = []string{"42.0.0.69"}
```

Comment:
this will work for HTTP but not for TCP because in TCP, we distinguish apps by port and not by the Host header. So, you might want to see if the Service Entry has only HTTP ports, and if so, do the DNS resolution, and if not, skip it altogether.

Also, for internaluse, we carved out 127.255.0.0/16 address space from which we mint VIPs and stick them into the service entries we generate. It works pretty well with Istio (you might want to do that in mcc).

Reply:
Yes, we should only be creating IPs out of the range 127.0.0.0/8 (or 127.255.0.0/16, w/e). Yes we know this is broken for TCP, but in TCP case we expect to have a VIP in hand anyway, it's only with a DNS entry for HTTP that we hit problems here.

Reply:
It was my understanding that it did: https://github.com/istio/istio/blob/8a84d6ddcadef0505f0678e99d65eff88ff0b496/tools/deb/istio-iptables.sh#L313

Reply:
Zack says the /32 is by design, so I've gone for the middle of 127.255.0.0/16. Ping @rshriram